releasing-archway.md

```diff
@@ -1,74 +1,41 @@
-    the emergency/critical fix from all other changes that have landed on the
-    release branch since the previous tag
-  - Cut release-X.Y.Z branch to hold the isolated patch release
-  - Tag release-X.Y.Z branch + fixes with X.Y.(Z+1)
-  - Branched [patch releases](#patch-releases) are rarely needed but used for
-    emergency/critical fixes to the latest release
-
-### Major version timeline
-
-There is currently no mandated timeline for major versions beyond version 1.Y.Z, Y,Z >= 0. We haven't so far applied a rigorous interpretation of semantic
-versioning with respect to incompatible changes of any kind.
-
-TODO: Major versioning criteria need to be put up for discussion once Archway protocol reaches 1.Y.Z release.
-
-## Patch releases
-
-Patch releases are intended for critical bug fixes to the latest minor version,
-such as addressing security vulnerabilities, fixes to problems affecting a large
-number of users and severe problems with no workaround.
-
-They should not contain miscellaneous feature additions or improvements, and
-especially no incompatibilities should be introduced between patch versions of
-the same minor version (or even major version).
-
-Dependencies, such as cosmos-sdk or tendermint, should also not be changed unless
-absolutely necessary, and also just to fix critical bugs (so, at most patch
-version changes, not new major nor minor versions).
+# Archway Protocol Release Process
+
+## Release Versioning
+
+This document outlines the release process for Archway protocol software.
+
+The **Archway** protocol follows a **state versioning system**, it does not utilize traditional [semantic versioning](http://semver.org); SemVer is specifically about APIs, but API breaking changes is not the only way in which libraries can "break". In deterministic systems, such as a blockchain being deterministic state machine, *state breaking changes far outweigh API breaking changes*. As such, Archway follows a State Versioning system.
+
+Given a version number Major.Minor.Patch, increment the:
+
+1. **Major** version when any state breaking changes are introduced;
+2. **Minor** version when any API changes, both API-compatible or API-incompatible, are introduced;
+3. **Patch** version when any state-compatible bug fixes are introduced;
+
+Additional labels for release-candidates, pre-release versions and other build metadata are available as extensions to the Major.Minor.Patch format.
+
+**Note:** Any dependency updates that are not state breaking, e.g. updating the Go version, fall under minor.
+
+## State Compatability
+
+**Note:** State breaking changes include changes that impact the amount of gas needed to execute a transaction as well as any changes to error handling. This is because `AppHash` and `LastResultsHash` contains:
+
+1. Tx `GasWanted`;
+2. Tx `GasUsed` - which is Merkelized, thus any logic affecting this will result in state changes;
+3. Tx response `Data` - protobuf encoding changes result in state changes;
```

Comment:
I think the msgs format in a transaction are also state breaking. So the Tx input as well as Tx response